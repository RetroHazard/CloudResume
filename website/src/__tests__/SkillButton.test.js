import React from 'react';
import { render } from '@testing-library/react';
import SkillButton from '../components/skill_button';

jest.mock('../utils/dataLoader', () => ({
    __esModule: true,
    default: ({ children }) =>
        children({
            target_skills: [
                { name: 'Skill 1', website: 'https://skill1.com', logo: 'skill1-icon' },
                { name: 'Skill 2', website: 'https://skill2.com', logo: 'skill2-icon' },
                // Add more mock skills as needed
            ]
        })
}));

describe('SkillButton', () => {
    it('renders skill buttons with correct data', () => {
        const skills = [
            { name: 'Skill 1', website: 'https://skill1.com', logo: 'skill1-icon' },
            { name: 'Skill 2', website: 'https://skill2.com', logo: 'skill2-icon' }
        ];
        const { asFragment } = render(<SkillButton skills={skills} />);

        // Check if skill buttons are rendered with correct data
        expect(asFragment()).toMatchSnapshot();
    });

    // Manual snapshot for review
    it('should match manual snapshot', () => {
        const skills = [
            { name: 'Skill 1', website: 'https://skill1.com', logo: 'skill1-icon' },
            { name: 'Skill 2', website: 'https://skill2.com', logo: 'skill2-icon' }
        ];
        const { asFragment } = render(<SkillButton skills={skills} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
