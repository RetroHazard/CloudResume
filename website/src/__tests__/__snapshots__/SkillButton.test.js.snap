// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillButton renders skill buttons with correct data 1`] = `
<DocumentFragment>
  <div
    class="flex flex-wrap max-sm:gap-1 sm:gap-2 md:gap-3"
  >
    <a
      class="skill-block"
      href="https://skill1.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <iconify-icon
        class="icon-box"
        classname="icon-box"
        height="1.1em"
        icon="skill1-icon"
        width="1.1em"
      />
      Skill 1
    </a>
    <a
      class="skill-block"
      href="https://skill2.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <iconify-icon
        class="icon-box"
        classname="icon-box"
        height="1.1em"
        icon="skill2-icon"
        width="1.1em"
      />
      Skill 2
    </a>
  </div>
</DocumentFragment>
`;

exports[`SkillButton should match manual snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-wrap max-sm:gap-1 sm:gap-2 md:gap-3"
  >
    <a
      class="skill-block"
      href="https://skill1.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <iconify-icon
        class="icon-box"
        classname="icon-box"
        height="1.1em"
        icon="skill1-icon"
        width="1.1em"
      />
      Skill 1
    </a>
    <a
      class="skill-block"
      href="https://skill2.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <iconify-icon
        class="icon-box"
        classname="icon-box"
        height="1.1em"
        icon="skill2-icon"
        width="1.1em"
      />
      Skill 2
    </a>
  </div>
</DocumentFragment>
`;
